src/test/java/ladder/domain/LineTest.java

```diff
@@ -1,22 +1,34 @@
 package ladder.domain;
 
+import ladder.domain.ladder.Bar;
+import ladder.domain.ladder.Line;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.DisplayName;
-import org.junit.jupiter.api.RepeatedTest;
+import org.junit.jupiter.api.MethodOrderer;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
 
 import java.util.List;
 
 import static org.assertj.core.api.Assertions.assertThat;
 
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
```

Comment:
우선 @TestInstance 어노테이션을 사용한 이유는
Line의 각 테스트들이 처음 생성된 List<Direction>을 수정하지 않기 때문에, 매 메서드를 실행 할 때마다 새 객체를 생성하는 것은 낭비라는 생각이 들었습니다.
하지만 각 테스트들이 서로 영향을 줘야하는 경우에 @TestInstance를 사용한다는 것은 알지만,

비결정적 테스트가 만들어지지 않음을 확실히 알고, 위와 같은 이유로 @TestInstance를 사용해도 문제가 될까요?

Reply:
조금 다른 맥락(불변 객체)에서 나온 얘기긴 하지만 오라클에서는 이렇게 말합니다 [링크](https://docs.oracle.com/javase/tutorial/essential/concurrency/immutable.html)

> Programmers are often reluctant to employ immutable objects, because they worry about the cost of creating a new object as opposed to updating an object in place. The impact of object creation is often overestimated, and can be offset by some of the efficiencies associated with immutable objects. These include decreased overhead due to garbage collection ...

코드를 구현할 때 너무 객체 생성의 효율성을 따지지 않아도 괜찮습니다. 실제로 이 어노테이션을 사용했을 때와 사용하지 않았을 때 걸리는 시간을 비교해보셔도 좋을 것 같아요.

객체 생성과 제거의 효율성은 gc에게 맡기고, 가독성/안정성 등에 조금 더 초점을 둔 코드를 작성해보시면 좋을 것 같습니다.

+) 만약 테스트에서 이 부분이 너무 느려서 조정이 필요하다면 물론 이런 방법들을 사용해서 속도를 빠르게 할 수도 있겠죠.